src/sonic-yang-models/yang-models/sonic-hash.yang

```diff
@@ -0,0 +1,67 @@
+module sonic-hash {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-hash";
+    prefix hash;
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "HASH YANG Module for SONiC OS";
+
+    revision 2022-09-05 {
+        description "First Revision";
+    }
+
+    typedef hash-field {
+        description "Represents native hash field";
+        type stypes:hash-field {
+            enum IN_PORT;
+            enum DST_MAC;
+            enum SRC_MAC;
+            enum ETHERTYPE;
+            enum VLAN_ID;
+            enum IP_PROTOCOL;
+            enum DST_IP;
+            enum SRC_IP;
+            enum L4_DST_PORT;
+            enum L4_SRC_PORT;
+            enum INNER_DST_MAC;
+            enum INNER_SRC_MAC;
+            enum INNER_ETHERTYPE;
+            enum INNER_IP_PROTOCOL;
+            enum INNER_DST_IP;
+            enum INNER_SRC_IP;
+            enum INNER_L4_DST_PORT;
+            enum INNER_L4_SRC_PORT;
+        }
+    }
+
+    container sonic-hash {
+
+        container SWITCH_HASH {
+
+            description "SWITCH_HASH part of config_db.json";
+
+            container GLOBAL {
+
+                leaf-list ecmp_hash {
```

Comment:
Why is this a leaf list. Do we expect more than one set of values?

Reply:
@dgsudharsan it is done for a convenient data representation:
```bash
root@sonic:/home/admin# sonic-cfggen --from-db --print | jq --indent 4 .SWITCH_HASH
{
    "GLOBAL": {
        "ecmp_hash": [
            "DST_MAC",
            "SRC_MAC",
            "DST_IP",
            "SRC_IP"
        ]
    }
}

root@sonic:/home/admin# redis-cli -n 4 HGETALL "SWITCH_HASH|GLOBAL"
1) "ecmp_hash@"
2) "DST_MAC,SRC_MAC,DST_IP,SRC_IP"
```

In this case user defines a list of elements in Config DB
and SONiC takes care for comma-separated representation in Config DB.

This improves config robustness, since it is harder to make a mistake in a list definition rather then in comma-separated string.